### PR TITLE
Handle overflow on nimble-list-option elements

### LINKED ISSUE #1658: Support for ellipsis text in dropdown list components
## 😯 Problem to Solve

Components like nimble-listbox, nimble-select, and nimble-combobox display a list of options for selection. These options can include long text, and if the client wishes to limit the width of the overall component, the text may become clipped or hidden within the option. To address this, ellipsis (...) can be used to indicate truncated text for long strings within the option.

Relevant conversation: https://github.com/ni/nimble/pull/1654#discussion_r1386962481

- [ ] Update the appropriate styles for nimble-combobox, nimble-select, nimble-listbox, and other components that utilize nimble-list-option
- [ ] Add Storybook matrix test coverage for all the aforementioned components
- [ ] Accessibility: If the ellipsis text is not visible, show a tooltip to display the full text

## 💁 Proposed Solution

Add [`text-overflow: ellipsis`](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow) to the `nimble-list-option` content slot element ([code link](https://github.com/ni/nimble/blob/47fc4c3463a2a2419d5a2be76e24b46eda90ff2f/packages/nimble-components/src/list-option/styles.ts#L29C5-L31C6)) like below:

``` CSS
.content {
    padding: 8px 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
```

### If client sets `max-width` or otherwise limits the width, text wraps
image

### If sufficient space is available, some components can grow but it eventually looks silly
image

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

Resolves #1658 

I also noticed that we had a workaround in the nimble-list-option for a FAST issue that has been fixed upstream, so I removed the workaround and validated that the issue is in fact fixed.

## 👩‍💻 Implementation

- Added styling to the nimble-list-option to ellipsis the content if it gets too long
- Leverage existing `overflow` directive to the list option template so that a `title` will be applied when the text is too long
    - This required forking the FAST template for the list option, so in addition to forking the template, I also brought in the FAST tests.

## 🧪 Testing

- Manually tested overflow behavior in storybook
- Manually tested that issue that had a workaround is still behaving as expected with the workaround removed.
- Wrote new unit tests to verify the overflow/title behavior
- Brought in the FAST unit tests for the list option since the template has now been forked

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by mollykreis** [comment]
@atmgrifter00, will you buddy this for me?

**Review by atmgrifter00 — APPROVED** [review]

File: change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: Could we have some storybook cases that exercise this?
The select and combobox are using our old style where you need the funky json editor to test things out. Could you instead create some useful cases we can try out with options?
Some ideas for option sets: simple (which has a couple disabled items), long list, wide list

Comment (rajsite) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: Guess it would also need a fixed width, currently it seems to just grow

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "patch",
```
Comment (jattasNI) [inline change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:2]: Let's update the matrix stories for affected components (select, combobox, listbox) to include an option that's truncated.

File: change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jattasNI) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: This only seems to be an issue in Firefox. In Chromium and Safari the text scrolls back to the start when you remove focus.

[the remark this answers is not in the document]
Comment (mollykreis) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: The styling to have the ellipsis on both sides is `text-overflow: ellipsis ellipsis`, but that styling is invalid in at least Chromium and causes no ellipsis to show up at all. I'm guessing we don't want to try to style the element differently based on browser?

[the remark this answers is not in the document]
Comment (rajsite) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: Fun css trick, invalid properties behave as if they are unset so something like the following might work (didn't try it):
```css
div {
    text-overflow: ellipsis;
    text-overflow: ellipsis ellipsis;
}
```
If we end up that way definitely needs a comment (maybe should be a csspartial if used in multiple spots)

[the remark this answers is not in the document]
Comment (rajsite) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: Looks like the two value version has very low compatibility, only firefox, so not sure it's great to use: https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow#browser_compatibility

[the remark this answers is not in the document]
Comment (mollykreis) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: I agree that it doesn't seem great to use, but I'm not sure how to proceed. Are we fine with keeping the current FF behavior as-is and not trying to change the behavior?

File: packages/nimble-components/src/list-option/template.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        class="${x => [x.checked && 'checked', x.selected && 'selected', x.disabled && 'disabled']
+        .filter(Boolean)
+        .join(' ')}"
+        role="option"
+    >
+        ${startSlotTemplate(context, definition)}
+        <span
+            class="content"
+            part="content"
+            ${overflow('hasOverflow')}
```
[the remark this answers is not in the document]
Comment (mollykreis) [reply packages/nimble-components/src/list-option/template.ts:37]: I don't see a lag with the 100-item list either. Do we have any expectations about the upper limit of items that should be put in the select or combo box?

File: change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jattasNI) [reply change/@ni-nimble-components-e81831d9-2db3-450d-b3a8-776ab5156e86.json:1]: Yep, I hadn't noticed the poor compatibility. Given that, I'm ok to drop this idea.

File: packages/nimble-components/src/list-option/template.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jattasNI) [reply packages/nimble-components/src/list-option/template.ts:37]: I don't recall any discussions or guidance about an upper limit. I can imagine a few hundred (e.g. countries or notebooks or users). Beyond that I suspect we'd advise clients to dynamically populate the options based on the filter the user has typed.

So manually testing with a hundred or so is probably sufficient. If we were talking hundreds of some larger resource like images or mutation observers maybe we'd need to be more careful.

**Review by jattasNI — APPROVED** [review]

**Pull request merged**